core/src/main/java/com/kin/ecosystem/core/util/ErrorUtil.java

```diff
@@ -74,16 +74,16 @@ public static KinEcosystemException fromApiException(ApiException apiException)
 						apiException);
 					break;
 				default:
-					exception = createUnknownException(apiException);
+					exception = createUnknownServiceException(apiException);
 					break;
 			}
 		}
 		return exception;
 	}
 
-	private static KinEcosystemException createUnknownException(@Nullable Throwable throwable) {
-		return new KinEcosystemException(KinEcosystemException.UNKNOWN, ECOSYSTEM_SDK_ENCOUNTERED_AN_UNEXPECTED_ERROR,
-			throwable);
+	private static KinEcosystemException createUnknownServiceException(@Nullable Throwable throwable) {
+		final String msg  = (throwable != null && throwable.getMessage() != null) ? throwable.getMessage() : ECOSYSTEM_SDK_ENCOUNTERED_AN_UNEXPECTED_ERROR;
```

Comment:
if throwable.getMessage() == null we might still get valuble inforamtion for example from getCause() - meaing we have undeling cause with more information.